### PR TITLE
Gradle 8 compatibility

### PR DESCRIPTION
**Pull request by evil159**
### What does this pull request do?

This PR makes plugin's build.gradle to be compatible with Gradle 8 by specifying the `namespace` attribute.
Also Example's app Gradle version is bumped to Gradle 8.

### What is the motivation and context behind this change?

https://github.com/mapbox/mapbox-maps-flutter/issues/217, https://mapbox.atlassian.net/browse/MAPSFLT-204

### Pull request checklist:
 - [x] Add a changelog entry.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
         Not applicable.
 - [ ] Add documentation comments for any added or updated public APIs.

### CODE REVIEW

**Review by pjleonard37 — APPROVED** [review]

File: android/build.gradle [hunk 1]
```diff
@@ -42,6 +42,7 @@ apply plugin: 'kotlin-android'
 android {
     compileSdk 33
 
+    namespace 'com.mapbox.maps.mapbox_maps'
```
Comment (jush) [inline android/build.gradle:45]: Where did you get this from?

File: example/android/settings.gradle [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version "8.3.1" apply false
```
Comment (jush) [inline example/android/settings.gradle:21]: I think there's also `8.3.2` and even `8.4.0` https://developer.android.com/build/releases/gradle-plugin

**Review by jush — APPROVED** [review]
LGTM but couple of small questions

File: android/build.gradle (hunk 1 above)
Comment (evil159) [reply android/build.gradle:45]: This was taken from the manifest file

File: example/android/settings.gradle (hunk 2 above)
Comment (evil159) [reply example/android/settings.gradle]: Bumped to `8.3.2`, I didn't notice. I keep it at `8.3` for Android Studio compatibility https://developer.android.com/build/releases/gradle-plugin#android_gradle_plugin_and_android_studio_compatibility

**Pull request merged**